src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4115,23 +4115,28 @@ private ParameterSyntax ParseParameter()
 
                 TypeSyntax type;
                 SyntaxToken name;
-                SyntaxToken exclamation = null;
+                SyntaxToken exclamationExclamation = null;
                 SyntaxToken equals = null;
                 if (this.CurrentToken.Kind != SyntaxKind.ArgListKeyword)
                 {
                     type = this.ParseType(mode: ParseTypeMode.Parameter);
                     name = this.ParseIdentifierToken();
                     if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
                     {
-                        exclamation = this.EatToken(SyntaxKind.ExclamationToken);
-                    }
-
-                    else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
-                    {
-                        var notEq = this.EatToken(SyntaxKind.ExclamationEqualsToken);
-                        equals = ConvertToMissingWithTrailingTrivia(notEq, SyntaxKind.EqualsToken);
-                        equals = AddError(equals, ErrorCode.ERR_NeedSpaceBetweenExclamationAndEquals);
-                        exclamation = SyntaxFactory.MissingToken(SyntaxKind.ExclamationToken);
+                        exclamationExclamation = this.EatToken();
+                        if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                        {
+                            // no trailing trivia and no leading trivia
+                            if (exclamationExclamation.GetTrailingTriviaWidth() == 0 && this.CurrentToken.GetLeadingTriviaWidth() == 0)
+                            {
+                                var exclamation2 = this.EatToken();
+                                exclamationExclamation = SyntaxFactory.Token(exclamationExclamation.GetLeadingTrivia(), SyntaxKind.ExclamationExclamationToken, exclamation2.GetTrailingTrivia());
+                            }
```

Comment:
I don't see any error reporting for this case?

Reply:
Moved to MergeTokens, will add error there.